bigquery/google/cloud/bigquery/job.py

```diff
@@ -1390,6 +1390,20 @@ def referenced_tables(self):
 
         return tables
 
+    @property
+    def num_dml_affected_rows(self):
```

Comment:
FYI: `numDmlAffectedRows` is one of few statistics that is also in QueryResults, which you added in https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2920

`totalBytesProcessed` and `cacheHit` also appear in QueryResults, but honestly I'd prefer people look at them on the Job resource.